[TASK] #100721 - Deprecate various language-related methods and properties (#2877)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/Localization/LocalizationApi/_LanguageService.rst.txt

   Main API to fetch labels from XLF (label files) based on the current system
   language of TYPO3. It is able to resolve references to files + their pointers to the
   proper language. If you see something about "LLL", this class does the trick for you. It
   is not related to language handling of content, but rather of labels for plugins.
   
   Usually this is injected into $GLOBALS['LANG'] when in backend or CLI context, and
   populated by the current backend user. Do not rely on $GLOBAL['LANG'] in frontend, as it is only
   available under certain circumstances!
   In frontend, this is also used to translate "labels", see TypoScriptFrontendController->sL()
   for that.
   
   As TYPO3 internally does not match the proper ISO locale standard, the "locale" here
   is actually a list of supported language keys, (see Locales class), whereas "English"
   has the language key "default".
   

>    Further usages on setting up your own LanguageService in BE:

Reviewer: ```suggestion
   Further usages on setting up your own language service in the backend:
```

   
   ..  code-block:: php
       

>        $languageService = GeneralUtility::makeInstance(LanguageServiceFactory::class)

Reviewer: Should we still use GeneralUtility here?

      
   .. php:attr:: debugKey
   
      If true, will show the key/location of labels in the backend.
      
      
   .. php:method:: getLL(string $index)
   

>       **Deprecated:** will be removed in TYPO3 v13.0. Use sL() instead.

Reviewer: can we link there?
